### DISCUSSION TITLE
Jobs error

### DISCUSSION THREAD

**Discussion by pLavrenov** [laravel/framework#33223] [General]
- Laravel Version: 7
- PHP Version: 7.4.6
- Database Driver & Version: MySQL 5.7.26
- Queue Driver: redis

When I send a task to the queue, I get an error.

`Error: Cannot unpack array with string keys in [PATH_TO_PROJ]/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Collection.php:605`

The array_intersect() function gets an array:
```
array:3 [
  "api_token" => []
  "api_client_id" => []
  "api_park_id" => []
]
```

Temporarily replaced the line and works:
`return array_intersect(...array_values($relations));`

Is this a bug or my mistake? Before updates everything worked.

----------------------------

[Database/Eloquent/Collection.php#L605](https://github.com/laravel/framework/blob/92d0de2d59bddb3251e281b1e9c767a123750139/src/Illuminate/Database/Eloquent/Collection.php#L605)

**Comment by driesvints** [comment]
We'll need some code here to reproduce.

**Comment by pLavrenov** [comment]
I was able to fully reproduce the chain
```
class User extends Model
{
    public function meta(): MorphMany
    {
        return $this->morphMany(Meta::class, 'metable');
    }

    public function setRelation($relation, $value)
    {
        if ($relation == 'meta') {
            $value = $value->keyBy('key');
        }
        return parent::setRelation($relation, $value);
    }
}
```
Получаем разницу в названия отношений.
![image](https://user-images.githubusercontent.com/11473265/84698572-b365d680-af58-11ea-9458-6627f12cd092.png)

```
class TestJob implements ShouldQueue
{
    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
    private $user;
    public function __construct(User $user)
    {
        $this->user = $user;
    }
    public function handle()   // Not fired 
    {
        dump(1);
    }
}
```

If you send such a model to the queue, it will give an error. But if you use array_values ​​() then it works in both cases.
